docs/references/http-gateway-protocol-spec.md

```diff
@@ -0,0 +1,471 @@
+# The HTTP Gateway Protocol Specification
+
+## Introduction
+
+The HTTP Gateway Protocol is an extension of the Internet Computer Protocol that allows conventional HTTP clients to interact with the Internet Computer network. This is important for software such as web browsers to be able to fetch and render client-side canister code, including HTML, CSS, and JavaScript as well as other static assets such as images or videos. The HTTP Gateway does this by translating between standard HTTP requests and requests that the Internet Computer Protocol will understand.
+
+Such a gateway could be a stand-alone proxy, it could be implemented in web browsers (natively, via a plugin or a service worker) or in other ways. This document describes the interface and semantics of this protocol independent of a concrete Gateway so that all Gateway implementations can be compatible.
```

Comment:
I would try to be consistent with capitalization:
```suggestion
Such a Gateway could be a stand-alone proxy, it could be implemented in web browsers (natively, via a plugin or a service worker) or in other ways. This document describes the interface and semantics of this protocol independent of a concrete Gateway so that all Gateway implementations can be compatible.
```